tools/dockerfile/grpc_artifact_protoc/Dockerfile

```diff
@@ -1,63 +0,0 @@
-# Copyright 2016 gRPC authors.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-# Docker file for building protoc and gRPC protoc plugin artifacts.
-# forked from https://github.com/google/protobuf/blob/master/protoc-artifacts/Dockerfile
-
-FROM centos:6.6
-
-RUN yum install -y git \
-                   tar \
-                   wget \
-                   make \
-                   autoconf \
-                   curl-devel \
-                   unzip \
-                   automake \
-                   libtool \
-                   glibc-static.i686 \
-                   glibc-devel \
-                   glibc-devel.i686
-
-# Install GCC 4.8
-RUN wget http://people.centos.org/tru/devtools-2/devtools-2.repo -P /etc/yum.repos.d
-RUN bash -c 'echo "enabled=1" >> /etc/yum.repos.d/devtools-2.repo'
-RUN bash -c "sed -e 's/\$basearch/i386/g' /etc/yum.repos.d/devtools-2.repo > /etc/yum.repos.d/devtools-i386-2.repo"
-RUN sed -e 's/testing-/testing-i386-/g' -i /etc/yum.repos.d/devtools-i386-2.repo
-
-# We'll get and "Rpmdb checksum is invalid: dCDPT(pkg checksums)" error caused by
-# docker issue when using overlay storage driver, but all the stuff we need
-# will be installed, so for now we just ignore the error.
-# https://github.com/docker/docker/issues/10180
-RUN yum install -y devtoolset-2-build \
-                   devtoolset-2-toolchain \
-                   devtoolset-2-binutils \
-                   devtoolset-2-gcc \
-                   devtoolset-2-gcc-c++ \
-                   devtoolset-2-libstdc++-devel \
```

Comment:
I think for protoc artifacts (since they are executables), we were linking libstdc++ statically to improve portability. Is that still the case?

Reply:
Yeah, we still keep it as is and these lines are for downloading gcc 4.8. (CentOS6 has 4.4 by default. [ref](https://www.youtube.com/watch?v=F1uT_oT5aow)).

From the investigation on generated artifacts using ldd, they're still statically linked with C++ runtimes.

```
$ ldd protoc
	linux-vdso.so.1 (0x00007fffcc71c000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fb93238b000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fb932246000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fb932225000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fb932062000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fb9323cc000)
$ ldd grpc_cpp_plugin 
	linux-vdso.so.1 (0x00007ffd11daa000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f00fd5bb000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f00fd5b0000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f00fd58f000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f00fd572000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f00fd42d000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f00fd26a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f00fd5e4000)
```